pandas/io/excel.py

```diff
@@ -66,17 +69,88 @@ def get_writer(engine_name):
         raise ValueError("No Excel writer '%s'" % engine_name)
 
 
+class BaseFile(object):
+    """ Class for identifying the type of reader
+    """
+
+    def __init__(self, try_engine=False):
+        if try_engine:
+            self.has_engine()
+
+    def is_ext(self, path):
+        """Verify if the path's extension is supported by the reader
+        """
+        ext = path.split('.')[-1]
+        if ext in self.extensions:
+            return True
+        else:
+            return False
+
+    def is_type(self, io):
+        """Verify if the io type is supported by the reader
+        """
+        if isinstance(io, self.io_class):
+            return True
+        else:
+            return False
+
+    def has_engine(self):
+        """Verify if the engine is installed
+        """
+        try:
+            self.load_engine()
+            _readers[self.engine] = True
+        except ImportError:
+            _readers[self.engine] = False
+
+
+class XLRDFile(BaseFile):
+
+    def __init__(self, **kwargs):
+        self.engine = 'xlrd'
+        self.extensions = ['xls', 'xlsx', 'xlsm']
+        self.io_class = type(None)
+        self.open_workbook = None
+        super(XLRDFile, self).__init__(**kwargs)
+
+    def load_engine(self):
```

Comment:
maybe define ``load_engine`` in ``BaseFile`` and have it raise ``NotImplementedError(...)`` to signal the sub-classes should impement